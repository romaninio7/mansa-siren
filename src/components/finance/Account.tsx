import * as React from 'react';
import { IAccount } from 'interfaces';
import { getCurrencySign, setAccountTypeColor, setDynamicArrow } from 'utils';

interface AccountProps extends IAccount {
  onAccountSelect: (id: string) => void;
  onOpenModal: (status: boolean) => void;
}

// A single account component
const Account = (props: AccountProps): JSX.Element => {
  const {
    account_id,
    account_type,
    account_number,
    currency,
    current,
    onAccountSelect,
    onOpenModal,
  } = props;

  //Handling a  click event
  const onClickHandle = (account_id: string, status: boolean): void => {
    onAccountSelect(account_id);
    onOpenModal(status);
  };

  return (
    <div
      className='card account-item'
      onClick={() => onClickHandle(account_id, true)}
    >
      <div
        className={`ui top right attached label account-type ${setAccountTypeColor(
          account_type
        )}`}
      >
        {account_type}
      </div>
      <div className='content'>
        <div className='header'>
          <i className={`${getCurrencySign(currency)} sign  icon`}></i>
          <span className='current'>{current}</span>
          <i
            className={`level ${setDynamicArrow(current)} alternate green icon`}
          ></i>
        </div>
        <div className='description'>
          <i className='icon credit card outline'></i> Account:&nbsp;
          <strong className='account-number'>{account_number}</strong>
        </div>
      </div>
      <div className='ui bottom attached button'>
        <i className='chart line icon'></i>
        Details
      </div>
    </div>
  );
};

export default Account;
