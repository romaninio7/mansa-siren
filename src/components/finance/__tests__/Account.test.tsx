import * as React from 'react';
import Account from 'components/finance/Account';
import { mount } from 'enzyme';

let wrapper;
const mockCallBack = jest.fn();
const mockCallBack2 = jest.fn();

describe('Account component rendering', () => {
  beforeEach(() => {
    wrapper = mount(
      <Account
        account_id='afcf2bd0'
        account_type='TRANSACTION'
        account_number='26178346'
        currency='GBP'
        current='175.52'
        onAccountSelect={mockCallBack}
        onOpenModal={mockCallBack2}
      />
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('has a valid snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shows a valid finance data', () => {
    expect(wrapper.find('.account-type').text()).toEqual('TRANSACTION');
    expect(wrapper.find('.account-number').text()).toEqual('26178346');
    expect(wrapper.exists('.pound')).toBeTruthy();
    expect(wrapper.find('.current').text()).toEqual('175.52');
  });

  // expect(mockCallBack.mock.calls.length).toEqual(1);

  it('fires valid functions callbacks', () => {
    expect(wrapper.find('.account-item').length).toEqual(1);
    wrapper.find('.account-item').simulate('click');
    expect(mockCallBack).toBeCalled();
    expect(mockCallBack2).toBeCalled();
  });
});
