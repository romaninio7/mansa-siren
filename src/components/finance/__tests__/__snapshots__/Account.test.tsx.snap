// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component rendering has a valid snapshot 1`] = `
<Account
  account_id="afcf2bd0"
  account_number="26178346"
  account_type="TRANSACTION"
  currency="GBP"
  current="175.52"
  onAccountSelect={[MockFunction]}
  onOpenModal={[MockFunction]}
>
  <div
    className="card account-item"
    onClick={[Function]}
  >
    <div
      className="ui top right attached label account-type purple"
    >
      TRANSACTION
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <i
          className="pound sign  icon"
        />
        <span
          className="current"
        >
          175.52
        </span>
        <i
          className="level up alternate green icon"
        />
      </div>
      <div
        className="description"
      >
        <i
          className="icon credit card outline"
        />
         Account:Â 
        <strong
          className="account-number"
        >
          26178346
        </strong>
      </div>
    </div>
    <div
      className="ui bottom attached button"
    >
      <i
        className="chart line icon"
      />
      Details
    </div>
  </div>
</Account>
`;
